main.py

```diff
@@ -20,235 +20,278 @@
 import pygame
-import sys
-from gettext import gettext as _
 
-from elements import *
-from welcomescreen import *
-from scorescreen import *
+from elements import Element
+from welcomescreen import Welcomescreen
+from scorescreen import Scorescreen
+
+# Enable use of activity with `python3 game.py`
+os.environ['SUGAR_BUNDLE_ID'] = 'org.sugarlabs.2cars'
+
+FPS = 60
+
+# Game states
+WELCOME = 0
+RULES = 1
+PLAY = 2
+PAUSE = 3
+
+BLACK = (0, 0, 0)
+WHITE = (255, 255, 255)
+
+# SPEED of cars in the x-axis, lane change speed
+ANGLE_LIMIT = 30
+ANGLE_SPEED = 2
+
+TICK_COUNTER_THRESHOLD = 40
+
+FONT_PATH_ARIMO = os.path.join("fonts", "Arimo.ttf")
+
+BACKGROUND_IMAGE_PATH = os.path.join("assets", "background.png")
+REDCAR_IMAGE_PATH = os.path.join("assets", "redcar.png")
+BLUECAR_IMAGE_PATH = os.path.join("assets", "bluecar.png")
+
+SOUND_HIT_PATH = os.path.join("assets", "sounds", "hit.ogg")
+SOUND_MISS_PATH = os.path.join("assets", "sounds", "miss.ogg")
+SOUND_MUSIC_PATH = os.path.join("assets", "sounds", "music.ogg")
+SOUND_SCORE_PATH = os.path.join("assets", "sounds", "score.ogg")
 
+RED_CIRCLE_PATH = os.path.join("assets", "redcircle.png")
+BLUE_CIRCLE_PATH = os.path.join("assets", "bluecircle.png")
+RED_SQUARE_PATH = os.path.join("assets", "redsquare.png")
+BLUE_SQUARE_PATH = os.path.join("assets", "bluesquare.png")
 
-class game:
 
+class Game:
     def __init__(self):
-        self.start = 1
+        self.state = WELCOME
+        pygame.font.init()
+        pygame.mixer.init()
+        self.load_assets()
+
+    def run(self):
+        self.initialize()
+        clock = pygame.time.Clock()
+        self.running = True
+        self.music.play(-1)
+        self.screen = pygame.display.get_surface()
+
+        while self.running:
+            self.render()
+            if self.state == WELCOME:
+                screen = Welcomescreen(self)
+                screen.run(self)
+                self.state = PLAY
+
+            self.handle_events()
+            self.update_game_state()
+
+            pygame.display.update()
+            clock.tick(FPS)
+        pygame.quit()
+        sys.exit()
+
+    def load_assets(self):
+        # Load Images
+        self.background_image = pygame.image.load(BACKGROUND_IMAGE_PATH)
+        self.leftcar_image = pygame.image.load(REDCAR_IMAGE_PATH)
+        self.rightcar_image = pygame.image.load(BLUECAR_IMAGE_PATH)
+        self.red_cricle_image = pygame.image.load(RED_CIRCLE_PATH)
+        self.blue_circle_image = pygame.image.load(BLUE_CIRCLE_PATH)
+        self.red_square_image = pygame.image.load(RED_SQUARE_PATH)
+        self.blue_square_image = pygame.image.load(BLUE_SQUARE_PATH)
+
+        # Load Music
+        self.hit = pygame.mixer.Sound(SOUND_HIT_PATH)
+        self.miss = pygame.mixer.Sound(SOUND_MISS_PATH)
+        self.music = pygame.mixer.Sound(SOUND_MUSIC_PATH)
+        self.scoresound = pygame.mixer.Sound(SOUND_SCORE_PATH)
+
+    # Scale images according to screen size, only happens once
```

Comment:
Redundant comment, although the comment says it's according to screen size but the scaling is littered with hard coded values.

Reply:
Fixed